### ISSUE TITLE
The Persistance example does not give the expected result

### ISSUE THREAD

**Issue by njoannin** [elasticsearch/elasticsearch-dsl-py#67]
Running the persistance example from the readme file (after small modifications, see #66), I checked the mappings of the blog index: they do not correspond to what was encoded in the example.

As a reminder, here is the Article class:

    class Article(DocType):
        title = String(analyzer='snowball', fields={'raw': String(index='not_analyzed')})
        body = String(analyzer='snowball')
        tags = String(index='not_analyzed')
        published_from = Date()
        lines = Integer()

And here is the resultant mapping in my instance of elasticsearch:

```
{
  "blog" : {
    "mappings" : {
      "article" : {
        "properties" : {
          "body" : {
            "type" : "string"
          },
          "lines" : {
            "type" : "long"
          },
          "published_from" : {
            "type" : "date",
            "format" : "dateOptionalTime"
          },
          "tags" : {
            "type" : "string"
          },
          "title" : {
            "type" : "string"
          }
        }
      }
    }
  }
}
```

As you can see, all fields with type: string are analyzed, whereas the title should be a multi-field and the tags field should not be analyzed.

Is this normal behavior?

Using: Python 3.3, elasticsearch-py 1.3.0, elasticsearch-dsl-py 0.0.3

**Comment by HonzaKral** [comment]
Thank you for the report, I forgot to re-test the code after making some changes.

The code should now work, this particular issue was caused by the example not callint `Article.init()` which would actually send the mappings into elasticsearch. Thus all the mappings were created implicitly by elasticsearch when a new article was saved.

**Comment by njoannin** [comment]
Thanks for the quick fix!
Cheers



Nicolas Joannin, Ph.D.
Independent research scientist
For hire on a per project basis

**Comment by arsham** [comment]
Hi,

I ran to the same problem with the same document type mentioned above:

```
{
    "blog": {
       "mappings": {
          "article": {
             "properties": {
                "body": {
                   "type": "string"
                },
                "lines": {
                   "type": "long"
                },
                "published_from": {
                   "type": "date",
                   "format": "dateOptionalTime"
                },
                "tags": {
                   "type": "string"
                },
                "title": {
                   "type": "string"
                }
             }
          }
       }
    }
}
```

My configuration is: Python 3.4.3, elasticsearch 1.4.0, elasticsearch-dsl 0.0.4 
Packages are installed with pip.

Thank you,
Arsham

**Comment by HonzaKral** [comment]
@arsham and what did you do, where is the code that you ran?

**Comment by arsham** [comment]
@HonzaKral Hi mate,

This is my model: 

```
class Article(DocType):

    title = String(
        analyzer='snowball',
        fields={
            'raw': String(index='not_analyzed')
        })

    body = String(analyzer='snowball')
    tags = String(index='not_analyzed')
    published_from = Date()
    lines = Integer()

    class Meta:
        index = 'blog'
```


Although it works fine if I create the mappings directly from Mapping API: 

```
mapping = Mapping('article')
mapping.field('title', 'string', fields={'raw': String(index='not_analyzed')})
mapping.field('body', 'string')
mapping.field('tags', 'string', index='not_analyzed')
mapping.field('published_from', 'date')
mapping.field('lines', 'integer')
mapping.save('blog')
```

**Comment by HonzaKral** [comment]
If you define the article you need to call `init` on it to actually create the mappings in the index or, preferably, use the `Index` class to create the index with the mappings - https://github.com/elastic/elasticsearch-dsl-py/blob/master/docs/persistence.rst#index

hope this helps

**Comment by arsham** [comment]
Thank you. It worked. Would you update the docs and explain you need to call the init on your new mappings please? I suppose the next person would run into the same issue. 

Cheers,
Arsham

**Comment by arsham** [comment]
Thank you it worked. 
Sorry I remove my last comment, I didn't notice you have already mentioned it in documents.

Cheers,
Arsham

**Comment by petercsmith** [comment]
Sorry to raise this again, but I'm at a complete loss for why my example code isn't working. I'm running the example code pasted below on Python 3.4.2 with elasticsearch 1.6.0 and elasticsearch-dsl 0.0.4. It doesn't seem to save and returns a 404 upon invoking get(). I've verified that the id is correct, indexes are being created on disk, es service is running, et al... it just doesn't seem to save! Your help is appreciated. 

from datetime import datetime
from elasticsearch_dsl import DocType, String, Date, Integer
from elasticsearch_dsl.connections import connections

# Define a default Elasticsearch client
connections.create_connection(hosts=['localhost'])

class Article(DocType):
    title = String(analyzer='snowball', fields={'raw': String(index='not_analyzed')})
    body = String(analyzer='snowball')
    tags = String(index='not_analyzed')
    published_from = Date()
    lines = Integer()

    class Meta:
        index = 'blog'

    def save(self, ** kwargs):
        self.lines = len(self.body.split())
        return super(Article, self).save(** kwargs)

    def is_published(self):
        return datetime.now() < self.published_from


# create the mappings in elasticsearch
Article.init()

# create and save and article
article = Article(id=42, title='Hello world!', tags=['test'])
article.body = ''' looong text '''
article.published_from = datetime.now()
article.save()

article = Article.get(id=42)
print(article.is_published())

# Display cluster health
print(connections.get_connection().cluster.health())

**Comment by HonzaKral** [comment]
This is fully and completely my fault - I verified the example in docs work (http://elasticsearch-dsl.readthedocs.org/en/latest/persistence.html#document-life-cycle) but completely forgot about the readme. I apologize, that is my mistake.

What happened here is that an `Article` was indeed created but the `id` given was just indexed as a field and not the document's ID - that was assigned by ES so the `get` couldn't find it. Proper way to create the `Article` is:

    article = Article(meta={'id': 42}, title='Hello world!', tags=['test'])

hope this helps. I will go fix the README now

**Comment by HonzaKral** [comment]
fixed as f48374aefb88978a430520648a6ec1bc27d7e0b6
